arrow-core-data/src/main/kotlin/arrow/core/set.kt

```diff
@@ -19,3 +22,36 @@ fun <A> Set<A>.eqv(EQA: Eq<A>, b: Set<A>): Boolean =
 fun <A> Set<A>.hashWithSalt(HA: Hash<A>, salt: Int): Int = HA.run {
   fold(salt) { hash, v -> v.hashWithSalt(hash) }
 }.hashWithSalt(size)
+
+fun <A> Semigroup.Companion.set(): Semigroup<Set<A>> =
+  Semigroup<Set<A>> { b: Set<A> ->
+    this + b
+  }
+
+fun <A> Monoid.Companion.set(): Monoid<Set<A>> = object : Monoid<Set<A>> {
+  override fun empty(): Set<A> =
+    emptySet<A>()
+
+  override fun Set<A>.combine(b: Set<A>): Set<A> =
+    this + b
+}
+
+fun <A> Eq.Companion.set(EQ: () -> Eq<A>): Eq<Set<A>> = object : Eq<Set<A>> {
+  override fun Set<A>.eqv(b: Set<A>): Boolean =
+    if (size == b.size) map { aa ->
+      b.find { bb -> EQ().run { aa.eqv(bb) } } != null
+    }.fold(true) { acc, bool ->
+      acc && bool
+    }
+    else false
+}
+
+fun <A> Show.Companion.set(SA: () -> Show<A>): Show<Set<A>> = object : Show<Set<A>> {
+  override fun Set<A>.show(): String =
+    show(SA())
+}
+
+fun <A> Hash.Companion.set(HA: Hash<A>): Hash<Set<A>> = object : Hash<Set<A>> {
```

Comment:
I've seen this done slightly different in the deprecation PR of map.. is preferable that way?
```
fun <K, A> Hash.Companion.map(HK: Hash<K>, HA: Hash<A>): Hash<Map<K, A>> =
  MapHash(HK, HA)

private class MapHash<K, A>(
  private val HK: Hash<K>,
  private val HA: Hash<A>
) : Hash<Map<K, A>> {
  override fun Map<K, A>.hashWithSalt(salt: Int): Int =
    hashWithSalt(HK, HA, salt)
}
```

Reply:
No, either way is fine :)